tchannel/peer_heap.py

```diff
@@ -80,3 +80,8 @@ def peek_peer(self):
             return None
 
         return self.peers[0]
+
+    def remove_peer(self, peer):
+        """Remove the peer from the heap."""
+        if peer.index < self.size():
```

Comment:
`peer.index` can be -1, right? change this to

    if 0 <= peer.index < self.size():